recipes/openssl/1.x.x/conanfile.py

```diff
@@ -223,15 +216,13 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
-
-    def layout(self):
-        pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        self.conf.define("tools.gnu:make_program", self._make_program)
```

Comment:
I asked the client experts about changing the conf, and the general advise is not to override the users input which is what should happen here.

In a future update I suspect we'll see a read only conf. The "define" functionality should be for the package info/conf info object to expose things in the package

Reply:
```suggestion
```

Reply:
Hi @prince-chrismc,
I removed the above statement as part of a larger refactoring to call nmake explicitly rather than leveraging Autotools.make().